src/iguana/algorithms/clas12/PhotonGBTFilter/Algorithm.cc

```diff
@@ -0,0 +1,418 @@
+          input_data.push_back(static_cast<float>(Epcal_ch[i]));
+          input_data.push_back(static_cast<float>(m2u_ch[i]));
+          input_data.push_back(static_cast<float>(m2v_ch[i]));
+      }
+
+      for (int i = 0; i < m_nh; ++i) {
+          input_data.push_back(static_cast<float>(R_nh[i]));
+          input_data.push_back(static_cast<float>(dE_nh[i]));
+          input_data.push_back(static_cast<float>(Epcal_nh[i]));
+          input_data.push_back(static_cast<float>(m2u_nh[i]));
+          input_data.push_back(static_cast<float>(m2v_nh[i]));
+      }
+
+      input_data.push_back(static_cast<float>(num_photons_0_1));
+      input_data.push_back(static_cast<float>(num_photons_0_2));
+      input_data.push_back(static_cast<float>(num_photons_0_35));
+
+      return ClassifyPhoton(input_data, runnum);
+  }
+
+  bool PhotonGBTFilter::ClassifyPhoton(std::vector<float> const &input_data, int const runnum) const
+  {
+
+      double sigmoid_x = 0.0; // Input to sigmoid
+      
+      // Assign the correct function based on runnum
+      if (runnum>=5032&&runnum<=5332) { // Fall2018 RGA Inbending
+          sigmoid_x = ApplyCatboostModel_RGA_inbending(input_data); 
+      } else if (runnum>=5333&&runnum<=5666) { // Fall2018 RGA Outbending
+          sigmoid_x = ApplyCatboostModel_RGA_outbending(input_data); 
+      } else if (runnum>=6616&&runnum<=6783) { // Spring2019 RGA Inbending
+          sigmoid_x = ApplyCatboostModel_RGA_inbending(input_data); 
+      } else if (runnum>=6156&&runnum<=6603) { // Spring2019 RGB Inbending
+          sigmoid_x = ApplyCatboostModel_RGA_inbending(input_data); 
+      } else if (runnum>=11093&&runnum<=11283) { // Fall2019 RGB Outbending
+          sigmoid_x = ApplyCatboostModel_RGA_outbending(input_data); 
+      } else if (runnum>=11284&&runnum<=11300) { // Fall2019 RGB BAND Inbending
+          sigmoid_x = ApplyCatboostModel_RGA_inbending(input_data); 
+      } else if (runnum>=11323&&runnum<=11571) { // Spring2020 RGB Inbending
+          sigmoid_x = ApplyCatboostModel_RGA_inbending(input_data); 
+      } else if (runnum>=16042&&runnum<=16772) { // Summer2022 RGC Inbending
+          sigmoid_x = ApplyCatboostModel_RGC_Summer2022(input_data); 
+      } else { // Unknown, defaulting to RG-A inbending
+          m_log->Warn("Run Number {} has no matching PhotonGBT model...Defaulting to RGA inbending...",runnum);
+          sigmoid_x = ApplyCatboostModel_RGA_inbending(input_data); 
+      }
+      
+      double prediction = 1-1/(1+exp(-sigmoid_x));               // Calculate predictive value for "signal"
+                                                                 // [0,1] --> Closer to 1 == Photon is Signal
+
+      return (prediction>o_threshold);
+  }
+  std::map<int, calo_row_data> PhotonGBTFilter::GetCaloMap(hipo::bank const& bank) const
+  {
+      std::map<int, calo_row_data> calo_map;
+      
+      for(int row = 0; row < bank.getRows(); row++){
```

Comment:
Comment added in 1c54f8c . I think it is best here to still loop over `bank.getRows()` to prevent upstream filters from affecting our filling of the calo map